src/github_api.rs

```diff
@@ -66,8 +66,13 @@ impl Client {
         }))
         .build();
 
+        let api_base = match hostname {
+            Some(hostname) => format!("https://{hostname}/api/v3"),
+            None => "https://api.github.com".to_string(),
+        };
+
         Self {
-            api_base: "https://api.github.com",
+            api_base: Box::leak(api_base.into_boxed_str()),
```

Comment:
No `Box::leak`, please! If this needs to be a `String` instead of a `&'static str`, that's perfectly fine.

Reply:
Will change that!